### ISSUE TITLE
Analytics Implementation or QA Support for Accredited Representation Management - Find a Representative

### ISSUE THREAD

**Issue by oddball-lindsay** [department-of-veterans-affairs/va.gov-team#70420]
# Google Analytics Implementation

### Required
- **Is this request for **Google Analytics implementation** or **QA**?**

     - [x] Google Analytics implementation
     - [x] QA

- **Who is this request for?**

    - Accredited Representation Management Team
    - PM = Lindsay Li-Smith
    
- **Provide the link(s) for Google Analytics implementation or QA**
  
    - Landing page: va.gov/get-help-from-accredited-representative
    - Find a Rep: va.gov/get-help-from-accredited-representative/find-rep
    - Product in Staging: https://staging.va.gov/get-help-from-accredited-representative/find-rep/
    
- **When is the launch date?**
    - Staged Rollout 3/5 - 3/18 ([full details](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/accredited-representation-management/product-documentation/find-a-representative/release-plan-find-a-representative-2.0.md#staged-rollout-details))

- **What interactions need to be tracked?** _NOTE: [Design System Components](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Design-System-Components.1840808040.html) are automatically tracked_
    
    - The "use my location" is a custom event.  We just need to track when this is being clicked, no need to track successes and failures.
    - Please see the KPIs in our [Find a Representative Product Outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/accredited-representation-management/product-documentation/find-a-representative/product-outline-find-a-representative.md#measuring-success) for additional context

- **If applicable, what test users are needed?**  _NOTE: Do not put staging credentials in this ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository_

    - Authentication is not required for this experience so no test users are needed

## Acceptance criteria
- [ ] All appropriate issue tagging is completed
- [ ] QA has been completed
- [ ] Events have been added to `Product` and `Benefit Hub` Content Groups

## Additional information

Please refer to [Platform Collaboration Cycle](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/index.html) or the [Analytics Request Touchpoint](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Analytics-request.1782120453.html) on Platform website for more information about the Collaboration Cycle.

**Comment by johnny-jesensky-adhoc** [comment]
This work will be carried over to the following sprint. This ticket has been carried over a few times, and would benefit from some more targeted AC, and a solid Definition of Done on this.

@jwoodman5 - I will work with the team on making the needed edits, and will plan to finalize this in the near future.

FYI @jestutt

**Comment by oddball-lindsay** [comment]
Some context that may be helpful around this history of this ticket:
I opened this when we were planning to launch our MVP of Find a Representative, however the team decided to "no go" after discovering that the representative contact information returned in the product's search results is not always accurate (and helpful to users).  

From there, our team has been working on a v2.0 that includes features to help address the inaccuracies with contact info.   This includes a banner above search results to inform users that not all info is up to date, and an option for users to report any contact information they find inaccurate.  We plan to monitor these reports and pass them along to the Office of General Counsel, who manages the database of rep contact information.

**We are aiming for Staging Review in a few weeks and hope to do a staged rollout in early March.**  

@johnny-jesensky-adhoc please let me know if I can help with anything in this particular ticket!  And if this "no go" scenario were to happen again, would it be better to close out the ticket and submit a new one closer to the new release date?

**Comment by johnny-jesensky-adhoc** [comment]
Hi @oddball-lindsay - thank you for all of this context. I just took over as PM for the Analytics Team, so that background is hugely helpful.

I think for now, we can keep this ticket open, and just keep it in our backlog until it's ready to be picked up. We can always edit the details if needed, or if the V2.0 is different from the original ask.

Feel free to provide updates as a comment on this ticket. Thanks!

**Comment by oddball-lindsay** [comment]
Hi @johnny-jesensky-adhoc we had our Staging Review yesterday and all went well, so we are on track for release!  Our Go/No Go is on 3/5 and if we decide to "Go", we will enter a staged rollout from 3/5 - 3/18 ([Staged Rollout Schedule](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/accredited-representation-management/product-documentation/find-a-representative/release-plan-find-a-representative-2.0.md#staged-rollout-schedule) if you're interested).

**Comment by oddball-lindsay** [comment]
@johnny-jesensky-adhoc we have an outstanding bug that is launch-blocking, so we will hold off on "Go" until this is resolved.  After resolution, staged rollout will begin and I can then predict the 100% live which will be ~2 weeks from the begin of staged rollout.

I'll circle back when we begin staged rollout so yall can predict when we'll hit 100% of users and determine from there the timing of analytics implementation :)

**Comment by jestutt** [comment]
@oddball-lindsay  Here are the dataLayer push events that the FEs need to add to the FAR search. I'll confirm them and the GTM/GA side once they are in place. Please let me know or have them reach out if they have any questions. 


### DataLayer Push Events

`far-search` fires when a new search is initiated. A new search is when either required fields are changed -- 'Type of accredited representative' and 'Address, city, state, or postal code' -- and hits search. 
```
'event': 'far-search'
'search-query': 'xxxxxxx', //full query input from the user -- i.e. 'Address, city, state, or postal code (*Required)'
'search-filters-list': {}, //key value pairs of search filter parameters like 'Type of accredied representative', 'Search area', 'Name of accredited representative'
'search-selection': 'Find VA Accredited Rep', 
```

`far-search-results` fires when the results of a new search loads for the first time. Should fire once for each search.
```
'event': 'far-search-results'
'search-query': 'xxxxxxx', //full query input from the user -- i.e. 'Address, city, state, or postal code (*Required)' 
'search-filters-list': {}, //key value pairs of search filter parameters like 'Type of accredied representative', 'Search area', 'Name of accredited representative' 
'search-selection': 'Find VA Accredited Rep', 
'search-results-total-count': 99, //populate with the total number of search results
'search-results-total-pages': 10, //populate with total number of result pages
```

`far-search-results-click` fires when a user clicks on any link in a result. 
```
'event': 'far-search-results-click'
'search-query': 'xxxxxxx', //full query input from the user -- i.e. 'Address, city, state, or postal code (*Required)' 
'search-filters-list': {}, //key value pairs of search filter parameters like 'Type of accredied representative', 'Search area', 'Name of accredited representative' 
'search-selection': 'Find VA Accredited Rep', 
'search-results-id': 'xxxxx', //populate with the individual result id
'search-results-total-count': 99, //populate with the total number of search results
'search-results-total-pages': 10, //populate with total number of result pages
'search-result-position': 99, //populate with the current pagination number
'search-result-page': 10, //populate with the current pagination number
```

`far-search-results-change` fires when a user only changes one of the non-required filter fields -- 'Search area' and 'Name of accredited representative' -- and hits search or changes the sort field -- 'Sort by' -- and hits apply. 
```
'event': 'far-search-results-change'
'search-query': 'xxxxxxx', //full query input from the user -- i.e. 'Address, city, state, or postal code (*Required)' 
'search-filters-list': {}, //key value pairs of search filter parameters like 'Type of accredied representative', 'Search area', 'Name of accredited representative' 
'search-selection': 'Find VA Accredited Rep', 
'search-results-change-action-type': 'sort', //dynamically populate with 'sort' or 'filter'
'search-results-change-action-label': 'newest', //'oldest' //populate according to the text label of user selection indicating the more specific action
'search-results-total-count': 99, //populate with the total number of search results
'search-results-total-pages': 10, //populate with total number of result pages
'search-results-page': 1, //populate with the current pagination number
```

`far-search-results-outdated` fires when a user clicks on a result's 'Report outdated inforamtion' button.
```
'event': 'far-search-results-outdated'
'search-query': 'xxxxxxx', //full query input from the user -- i.e. 'Address, city, state, or postal code (*Required)' 
'search-filters-list': {}, //key value pairs of search filter parameters like 'Type of accredied representative', 'Search area', 'Name of accredited representative' 
'search-selection': 'Find VA Accredited Rep', 
'search-results-id': 'xxxxx', //populate with the individual result id
'search-results-total-count': 99, //populate with the total number of search results
'search-results-total-pages': 10, //populate with total number of result pages
'search-result-position': 99, //populate with the current pagination number
'search-result-page': 10, //populate with the current pagination number
```

Not that the search-results-id referred to in the `far-search-results-click` and `far-search-results-outdated` events is based on the id I can see in the HTML (screenshot below): 
<img width="1270" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBenU2QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--259812c868812b7270401d51b71ce724c4d692af/Screenshot%202024-03-05%20at%201.19.48%E2%80%AFPM.png" alt="Screenshot 2024-03-05 at 1.19.48 PM.png" />


### Additional Items

- The FEs were wonderful and used components to build the search results list but we want to avoid multiple events firing for the same interaction. To do this, the [va-button](https://design.va.gov/storybook/?path=/docs/components-va-button--primary) and [va-telephone](https://design.va.gov/storybook/?path=/docs/components-va-telephone--default) components need to include the `disable-analytics=true` property setting.  
- The dataLayer push event for the 'Report outdated information' modal is incomplete (screenshot below). At least the `modal-title` need to be included though we don't want to include the name -- may just use 'Report outdated information' so we can identify the type of modal. 

<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMW02QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--543c999dd7f963f2eea897f284f9c615f2e002b7/Screenshot%202024-03-06%20at%2011.33.50%E2%80%AFAM.png" alt="Screenshot 2024-03-06 at 11.33.50 AM.png" />

**Comment by oddball-lindsay** [comment]
@jestutt PR ^ for that datalayer work (cc @cosu419)

**Comment by oddball-lindsay** [comment]
@jestutt I think all the work you need has been completed with the merged PRs above.  Do you or @johnny-jesensky-adhoc have an estimate of when this request will be completed, with setup on the GTM/GA side?

I'm planning to start discussing the Domo Dashboard needs after this is complete.

**Comment by jestutt** [comment]
@oddball-lindsay  I'm trying to get it into this sprint but it might it might go into the beginning of the next. We're likely going to need custom dimensions for some of the search fields -- same method as tracking on other searches -- which makes this a little more complicated than setting up for other events. Will let you know when we start working on them.

**Comment by jestutt** [comment]
@oddball-lindsay  I've been working on the GTM implementation and have noticed a few issues with the events. 

- When I initiate the first search, an extra, incomplete `far-search-results` fires before the real (?)  `far-search` and `far-search-results` events. I think I can exclude this extra event from being collected if it can't be removed. ![Screenshot 2024-05-05 at 6 24 01 PM](https://github.com/department-of-veterans-affairs/va.gov-team/assets/141653184/0bf1d41a-6337-42d2-ab4a-4919d8ea201b)

- The `search-filters-list` variables for the `far-search` event aren't formatted correctly -- it's normal language instead of the dashed version, ex "Type of accredited representative instead" of `representative-type`. ![Screenshot 2024-05-05 at 6 23 43 PM](https://github.com/department-of-veterans-affairs/va.gov-team/assets/141653184/f9c56264-d7fb-4435-b075-83b651266887)

- `far-search-results-change` is not firing at all. Instead, a new `far-search` and `far-search-results` events pair fires when the filters are changed and nothing fires on sort. 

- The `far-search-results-click` and `far-search-results-outdated` events do not have any values in the `search-results-total-count`, `search-results-total-pages`. `search-result-position` and 
- `search-result-page` variables.  ![Screenshot 2024-05-05 at 6 23 20 PM](https://github.com/department-of-veterans-affairs/va.gov-team/assets/141653184/b1b67241-45e4-42c3-82cd-9eb6af12a2e1)

I should have the GTM implementation completed today, I just won't be able to check the change events tracking at this time.

**Comment by oddball-lindsay** [comment]
Thanks for flagging these @jestutt!  We are mid-sprint at the moment, but can pick up these changes in our next sprint which is 5/14-5/27.

**Comment by jestutt** [comment]
@oddball-lindsay  I've put together the UA and GA4 events in GTM but haven't been able to confirm them at the moment since it looks like your team is currently working on them. Rather than wait, I've gone ahead and published the GTM tags and we can refine and confirm once the devs have completed their update. 

@johnny-jesensky-adhoc We can move this to a refinement/confirmation ticket in the next sprint and @Blivaditis will be able to assist from here.

**Comment by johnny-jesensky-adhoc** [comment]
Great -- @oddball-lindsay, as Jamie alluded to, and to better track the work for our team on this -- I'm going to close this ticket out. We can look to create a separate task for additional refinement needed on this, once your team is done with their work on these events, and has more bandwidth.

FYI @Blivaditis , @jestutt
